### PR TITLE
Migrate to async/await

### PR DESCRIPTION
**Pull request by anthonyringoet**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lib/watch/fork.js [hunk 1]
```diff
@@ -35,7 +32,18 @@ process.on('uncaughtException', (err) => {
...
+
+  try {
+    await fastify.ready()
+    process.send({ type: READY })
+  } catch (err) {
+    stop(err)
+  }
 }
+
+main()
```
Comment (mcollina) [inline lib/watch/fork.js:49]:
```suggestion
main().catch((err) => {
  console.error(err)
  process.exit(1)
})
```

Comment (anthonyringoet) [reply lib/watch/fork.js:49]: 👍

**Comment by anthonyringoet** [comment]
Rebase pushed.

**Comment by anthonyringoet** [comment]
Should `should start the server with watch options that the child process restart when directory changed` be skipped in Github Actions CI? I presume it was using Travis before (hence the `onTravis` skip filter).

File: test/start.test.js [hunk 2]
```diff
@@ -612,8 +565,8 @@ test('crash on unhandled rejection', t => {
 })
 
 if (!process.version.match(/v[0-6]\..*/g)) {
```
Comment (mcollina) [inline test/start.test.js:567]: I think this if can be removed.

**Comment by mcollina** [comment]
Can you please add a link to that if? I can't quickly find it. Overall if it was skipped before it should be skipped now.

File: test/print-routes.test.js [hunk 3]
```diff
@@ -96,13 +97,14 @@ test('should print routes of server with an async/await plugin', t => {
...
-      t.ok(spy.called)
-      t.ok(spy.calledWithMatch('debug', '└── / (GET)\n'))
-    })
-  })
+    await fastify.close()
+    t.ok(spy.called)
+    t.ok(spy.calledWithMatch('debug', '└── / (GET)\n'))
+  } catch (err) {
+    if (err) t.error(err)
+  }
```
Comment (mcollina) [inline test/print-routes.test.js:109]: Please remove this try-catch, it's not needed.

File: test/print-routes.test.js [hunk 4]
```diff
@@ -8,22 +8,23 @@ const printRoutes = require('../print-routes')
...
-      t.ok(spy.called)
-      t.ok(spy.calledWithMatch('debug', '└── / (GET|POST)\n'))
-    })
-  })
+    await fastify.close()
+    t.ok(spy.called)
+    t.ok(spy.calledWithMatch('debug', '└── / (GET|POST)\n'))
+  } catch (err) {
+    t.error(err)
+  }
```
Comment (mcollina) [inline test/print-routes.test.js:27]: remove the try-catch

**Comment by anthonyringoet** [comment]
@mcollina FYI [`onTravis` skip](https://github.com/fastify/fastify-cli/blob/1acb82725a43832ad1a99d2a9e6dfd4a9293ff9b/test/start.test.js#L571-L602), just [pushed replacement](https://github.com/fastify/fastify-cli/pull/253/commits/b38ca156cd5d4bb2892f8f418ea36679a3aa15ad) to detect Github Actions CI environment.

Also cleaned up the try/catch statements and node 6 conditional.

**Comment by mcollina** [comment]
Good work!

**Pull request merged**